src/test/java/com/google/cloud/genomics/dataflow/utils/VariantUtilsTest.java

```diff
@@ -34,7 +34,16 @@
   public void testIsVariant() {
     assertTrue(VariantUtils.isVariant(DataUtils.makeVariant("chr7", 200000, 200001, "A",
```

Comment:
My bad - it looks like indeed, isVariant will return true for [A->G].

I think the bottom line is that I'm confused. What is the purpose of
"isVariant"? It takes in a Variant object, so naively one might think that
every Variant object surely must contain a variant. What else could it be?
I don't know what a block record is. I *think* you're talking about the
<NON_REF> 0/0 calls which are used to indicate regions where we have
coverage but where there is no variation. Is that what it is?

It would be helpful to have a javadoc above "isVariant" that explains the
function's purpose.


On Tue, May 5, 2015 at 3:58 PM, Nicole Deflaux <notifications@github.com>
wrote:

> In
> src/test/java/com/google/cloud/genomics/dataflow/utils/VariantUtilsTest.java
> <https://github.com/googlegenomics/dataflow-java/pull/63#discussion_r29722224>
> :
>
> > @@ -34,7 +34,16 @@
> >    public void testIsVariant() {
> >      assertTrue(VariantUtils.isVariant(DataUtils.makeVariant("chr7", 200000, 200001, "A",
>
> Thanks for taking a look.
>
> @jean-philippe-martin <https://github.com/jean-philippe-martin> Here is
> the test for SNPs. isVariant returns true. I will re-write isVariant to be
> more readable.
>
> @pgrosu <https://github.com/pgrosu> the purpose of that method is to
> differentiate between variants and non-variant segments (a.k.a. non-variant
> block records). For Complete Genomics data and gVCFs such as Platinum
> Genomes, we wind up with zero alternates (the missing value alternate .
> in the VCF spec gets converted to null). That reminds me, I should add some
> logic for <NON_REF> which is emitted by
> https://www.broadinstitute.org/gatk/guide/article?id=4017
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/googlegenomics/dataflow-java/pull/63/files#r29722224>.
>